one_fm/operations/doctype/contracts/contracts.py

```diff
@@ -1028,56 +1028,58 @@ def send_contract_reminders(is_scheduled_event=True):
-            get_date_str(i.contract_termination_decision_period_date),i.name,get_date_str(i.start_date),get_date_str(i.contract_end_internal_notification_date),\
-            get_date_str(i.end_date),i.duration,i.client,i.engagement_type, i.contract] for i in contracts_due_internal_notification]
-        for each in contracts_due_internal_notification_list:
-            context = {"project": each[8],
-                       "contract_end_internal_notif_period": each[1],
-                       "start_date": each[4],
-                       "engagement_type": each[9],
-                       "contract_end_internal_notif_date": each[5],
-                       "contract_termination_decision_period": each[0],
-                       "contract_termination_decision_date": each[2],
-                       "end_date": each[6],
-                       "duration": each[7],
-                       "document_id": each[3],
-                       "link": frappe.utils.get_url_to_form("Contracts",each[3]),
-                       "attachment": frappe.utils.get_url(each[10]) if each[10] else None}
-            msg = frappe.render_template('one_fm/templates/emails/contracts_reminder.html', context=context)
-            sendemail(recipients=[users], subject="Expiring Contracts", content=msg, is_scheduler_email=is_scheduled_event)
-    if contracts_due_termination_notification:
-        contracts_due_termination_notification_list = [[i.contract_termination_decision_period,i.contract_end_internal_notification,\
-            get_date_str(i.contract_termination_decision_period_date),i.name,get_date_str(i.start_date),get_date_str(i.contract_end_internal_notification_date),\
-            get_date_str(i.end_date),i.duration,i.client,i.engagement_type, i.contract] for i in contracts_due_termination_notification]
-        for each in contracts_due_termination_notification_list:
-            context = {"project": each[8],
-                       "contract_end_internal_notif_period": each[1],
-                       "start_date": each[4],
-                       "engagement_type": each[9],
-                       "contract_end_internal_notif_date": each[5],
-                       "contract_termination_decision_period": each[0],
-                       "contract_termination_decision_date": each[2],
-                       "end_date": each[6],
-                       "duration": each[7],
-                       "document_id": each[3],
-                       "link": frappe.utils.get_url_to_form("Contracts",each[3]),
-                       "attachment": frappe.utils.get_url(each[10]) if each[10] else None}
-            msg = frappe.render_template('one_fm/templates/emails/contracts_reminder.html', context=context)
-            sendemail(recipients=[users], subject="Expiring Contracts", content=msg,is_scheduler_email=is_scheduled_event)
+    try:
```

Comment:
Please what change did you make

Reply:
I wrap everything in try and catch and I also pass users variable into the sendemail function directly